src/black/__init__.py

```diff
@@ -602,6 +612,18 @@ def main(
     ctx.exit(report.return_code)
 
 
+def test_regex(
+    ctx: click.Context,
+    regex_name: str,
+    regex: Optional[str],
+) -> Optional[Pattern]:
+    try:
+        return re_compile_maybe_verbose(regex) if regex is not None else None
+    except re.error:
+        err(f"Invalid regular expression for {regex_name} given: {regex!r}")
+        ctx.exit(2)
```

Comment:
The more I look at this code, perhaps this would be better handled via click argument callbacks? Then, main() would either get None or a compiled regex right from the get go. IDK and IDC that much though

If that would be an improvement, it should probably be handled in another PR.

Reply:
I was thinking that when making the change, but ultimately figured it was this way on purpose 😄.

I'll swing back around for this if no one beats me to it.

Reply:
I went ahead and made a PR on top of this branch. I'll switch branches when this PR is in.

https://github.com/thejcannon/black/pull/1

Reply:
Something tells me I won't be able to switch _repos_ for that PR though. Feel free to comment on it, or wait until I republish and target this repo.

Reply:
Done and dusted: https://github.com/psf/black/pull/2016